### ISSUE TITLE
Are there plans for a `flang-format`?

### ISSUE THREAD

**Issue by knedlsepp** [flang-compiler/flang#250]
Dealing with a crufty Fortran codebase I've often times wished for some tool to help in unifying the formatting style.
I have no idea about the magnitude of such work, but I'd really like to know if there are any plans for implementing a tool similar to [clang-format](https://clang.llvm.org/docs/ClangFormat.html). (or [YAPF](https://github.com/google/yapf) for python)
Could a potential implementation reuse parts of this codebase? How about the generation of the AST?

**Comment by gklimowicz** [comment]
You're right, this would be nice, but we don't have any plans at this time for a flang-format command.

**Comment by pbregener** [comment]
I think this would not only be nice, but a huge step for the Fortran community! Especially given the loads of legacy code (that often went through automatic fixed-form-to-free-form conversions and alike) in the wild and the evolution of coding styles and guidelines over the last decades, Fortran could really benefit a lot from such a tool and flang-format might be an effective gateway drug for flang, too.

Similar to the OP, I have no idea about the magnitude of the required effort, but I'd also be really interested to know what is actually needed to "turn clang-format into flang-format".

**Comment by williamfgc** [comment]
there is a similar effort: https://github.com/pseewald/fprettify
